packages/react/src/components/table/utils/table-utils.ts

```diff
@@ -24,27 +24,44 @@ export function calculateStickyColumns(
 
 export function calculateStickyHeader(
     stickyColumns: boolean[],
-    headerCells: HTMLCollectionOf<HTMLTableHeaderCellElement>,
+    headerCells: NodeListOf<HTMLTableHeaderCellElement>,
 ): void {
     Array.from(headerCells).forEach((headerCell, index) => {
         headerCell.style.setProperty('top', '0px');
         headerCell.style.setProperty('z-index', stickyColumns[index] ? '5' : '4');
     });
 }
 
+export function calculateStickyFooter(
+    stickyColumns: boolean[],
+    footerCells: NodeListOf<HTMLTableCellElement>,
+): void {
+    Array.from(footerCells).forEach((footerCell, index) => {
+        footerCell.style.setProperty('bottom', '0px');
+        footerCell.style.setProperty('z-index', stickyColumns[index] ? '5' : '4');
+    });
+}
+
 export function calculateStickyPosition(
     stickyColumns: boolean[],
     stickyHeader: boolean,
+    stickyFooter: boolean,
     tableRef: React.RefObject<HTMLTableElement>,
 ): void {
     if (tableRef.current === null) {
         return;
     }
-    const headerCells = tableRef.current.getElementsByTagName('th');
-    const rows = tableRef.current.getElementsByTagName('tr');
+    const headerCells = tableRef.current?.querySelectorAll('th');
+    const rows = tableRef.current.querySelectorAll('tr');
```

Comment:
The line above use `tableRef.current?`, make it the same, if both need "?" or not.